File: backend/src/main/java/org/example/backend/domain/member/dto/MemberEditRequestDto.java
package org.example.backend.domain.member.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record MemberEditRequestDto(
    @NotBlank
    @Email
    String email,

    @NotBlank
    @Size(min = 8, max = 20)
    String currentPassword, // 현재 비밀번호 (필수)

    @NotBlank
    @Size(min = 8, max = 20)
    String newPassword, // 새로운 비밀번호

    @NotBlank
    String nickname,

    String profileImageUrl
) {}

File: backend/src/main/java/org/example/backend/domain/member/dto/MemberEditRequestDto.kt
package org.example.backend.domain.member.dto

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class MemberEditRequestDto(
    @field:NotBlank @field:Email val email: String?,
    @field:NotBlank @field:Size(min = 8, max = 20) val currentPassword: String?,  // 현재 비밀번호 (필수)
    @field:NotBlank @field:Size(min = 8, max = 20) val newPassword: String?,  // 새로운 비밀번호
    @field:NotBlank val nickname: String?,
    val profileImageUrl: String?
)
